extensions/cl_ext_float_atomics.asciidoc

```diff
@@ -0,0 +1,729 @@
+C atomic_fetch_max(volatile __local A *object, M operand)
+C atomic_fetch_min_explicit(volatile __local A *object, M operand, memory_order order)
+C atomic_fetch_max_explicit(volatile __local A *object, M operand, memory_order order)
+C atomic_fetch_min_explicit(volatile __local A *object, M operand,
+    memory_order order, memory_scope scope)
+C atomic_fetch_max_explicit(volatile __local A *object, M operand,
+    memory_order order, memory_scope scope)
+
+// In addition to the requirements described in the OpenCL C 3.0 specification,
+// requires the __opencl_c_ext_fp16_global_atomic_min_max feature
+// and the __opencl_c_ext_fp16_local_atomic_min_max feature (for atomic_half),
+// requires the __opencl_c_ext_fp32_global_atomic_min_max feature
+//and the __opencl_c_ext_fp32_local_atomic_min_max feature (for atomic_float), or
+// requires the __opencl_c_ext_fp64_global_atomic_min_max feature
+// and the __opencl_c_ext_fp64_local_atomic_min_max feature (for atomic_double).
+C atomic_fetch_min(volatile A *object, M operand)
+C atomic_fetch_max(volatile A *object, M operand)
+C atomic_fetch_min_explicit(volatile A *object, M operand, memory_order order)
+C atomic_fetch_max_explicit(volatile A *object, M operand, memory_order order)
+C atomic_fetch_min_explicit(volatile A *object, M operand,
+    memory_order order, memory_scope scope)
+C atomic_fetch_max_explicit(volatile A *object, M operand,
+    memory_order order, memory_scope scope)
+----
+
+== Modifications to the OpenCL API Specification
+
+Add to Table 5 - OpenCL Device Queries in Section 4.2 - Querying Devices: ::
++
+--
+[caption="Table 5. "]
+.List of supported param_names by clGetDeviceInfo
+[width="100%",cols="4,3,5",options="header"]
+|====
+| Device Info | Return Type | Description
+| {CL_DEVICE_SINGLE_FP_ATOMIC_CAPABILITIES_EXT} +
+  {CL_DEVICE_DOUBLE_FP_ATOMIC_CAPABILITIES_EXT} +
+  {CL_DEVICE_HALF_FP_ATOMIC_CAPABILITIES_EXT} +
+  | {cl_device_fp_atomic_capabilities_ext_TYPE}
+      | Describes the floating-point atomic operations supported by the device.
+        This is a bit-field that describes a combination of the following values:
+
+        `CL_DEVICE_GLOBAL_FP_ATOMIC_LOAD_STORE_EXT` - Can perform floating-point load, store, and exchange atomic operations in global memory. +
+        `CL_DEVICE_GLOBAL_FP_ATOMIC_ADD_EXT` - Can perform floating-point addition and subtraction atomic operations in global memory. +
+        `CL_DEVICE_GLOBAL_FP_ATOMIC_MIN_MAX_EXT` - Can perform floating-point min and max atomic operations in global memory.
+
+        `CL_DEVICE_LOCAL_FP_ATOMIC_LOAD_STORE_EXT` - Can perform floating-point load, store, and exchange atomic operations in local memory. +
+        `CL_DEVICE_LOCAL_FP_ATOMIC_ADD_EXT` - Can perform floating-point addition and subtraction atomic operations in local memory. +
+        `CL_DEVICE_LOCAL_FP_ATOMIC_MIN_MAX_EXT` - Can perform floating-point min and max atomic operations in local memory.
```

Comment:
Good point, I didn't intend for all queries to be be `0` at once.  Your suggested sentence makes sense.

Reply:
After thinking about this a bit more I've taken your suggestion as-is.  I think it would be a little strange to support the extension and return zero for all of the queries but perhaps there's a reason for doing so.  Regardless, it seems like an extreme corner case we're unlikely to encounter in practice.